src/OpenApi/src/Services/OpenApiDocumentService.cs

```diff
@@ -68,12 +106,20 @@ private OpenApiPaths GetOpenApiPaths(HashSet<OpenApiTag> capturedTags)
         return paths;
     }
 
-    private static Dictionary<OperationType, OpenApiOperation> GetOperations(IGrouping<string?, ApiDescription> descriptions, HashSet<OpenApiTag> capturedTags)
+    private Dictionary<OperationType, OpenApiOperation> GetOperations(IGrouping<string?, ApiDescription> descriptions, HashSet<OpenApiTag> capturedTags)
     {
         var operations = new Dictionary<OperationType, OpenApiOperation>();
         foreach (var description in descriptions)
         {
-            operations[description.GetOperationType()] = GetOperation(description, capturedTags);
+            var operation = GetOperation(description, capturedTags);
+            operation.Extensions.Add(OpenApiConstants.DescriptionId, new OpenApiString(description.ActionDescriptor.Id));
+            OperationTransformerContextCache.TryAdd(description.ActionDescriptor.Id, new OpenApiOperationTransformerContext
+            {
+                DocumentName = documentName,
+                Description = description,
+                ApplicationServices = serviceProvider,
+            });
+            operations[description.GetOperationType()] = operation;
```

Comment:
Is clobbering okay?  If not, is there some reason we believe it can't happen?

Reply:
Clobbering is not likely to happen here. Operations represent a unique combination of HTTP method + path and since the routing layer already warns about duplicate endpoints (same HTTP method and path) it's unlikely you'll encounter the same operation twice.